src/Text/Pandoc/ImageSize.hs

```diff
@@ -116,7 +116,11 @@ imageType img = case B.take 4 img of
                      _                  -> mzero
 
 findSvgTag :: ByteString -> Bool
-findSvgTag img = B.null $ snd (B.breakSubstring img "<svg")
+findSvgTag img = prfx `elem` ["<svg ", "<svg\n", "<svg\r"]
+  where
+    prfx = B.take 5 $ last $ B.groupBy openingTag $ B.drop 7 img
+    -- B.groupBy openingTag matches first "<svg" or "<html" but not "<!--"
+    openingTag x y = x == '<' && y /= '!'
```

Comment:
@jgm true, so we're now at:

    findSvgTag img = snip `elem` ["<svg ", "<svg\n", "<svg\r"]
      where
        snip = B.take 5 $ lastOrEmpty $ B.groupBy openingTag img
        openingTag x y = x == '<' && not (y == '!' || y == '?')
        lastOrEmpty xs = foldl (\_ x -> x) [] xs

about adding tests: maybe add a file to `/test/command` that tries to load `/test/command/svg-test.svg` etc..?